fenix/app/src/androidTest/java/org/mozilla/fenix/ui/DownloadTest.kt

```diff
@@ -238,4 +244,337 @@ class DownloadTest {
+        }.clickDownload {
+            verifyDownloadedFileName(secondDownloadedFile)
+        }
+        browserScreen {
+        }.openThreeDotMenu {
+        }.openDownloadsManager {
+            verifyDownloadedFileName(firstDownloadedFile)
+            verifyDownloadedFileName(secondDownloadedFile)
+            longClickDownloadedItem(firstDownloadedFile)
+            selectDownloadedItem(secondDownloadedFile)
+            openMultiSelectMoreOptionsMenu()
+            clickMultiSelectRemoveButton()
+            verifyEmptyDownloadsList()
+        }
+    }
+
+    @Test
+    fun undoDeleteMultipleDownloadedFilesTest() {
+        val firstDownloadedFile = "smallZip.zip"
+        val secondDownloadedFile = "textfile.txt"
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(downloadTestPage.toUri()) {
+            waitForPageToLoad()
+        }.clickDownloadLink(firstDownloadedFile) {
+            verifyDownloadPrompt(firstDownloadedFile)
+        }.clickDownload {
+            verifyDownloadedFileName(firstDownloadedFile)
+        }.closeCompletedDownloadPrompt {
+        }.clickDownloadLink(secondDownloadedFile) {
+            verifyDownloadPrompt(secondDownloadedFile)
+        }.clickDownload {
+            verifyDownloadedFileName(secondDownloadedFile)
+        }
+        browserScreen {
+        }.openThreeDotMenu {
+        }.openDownloadsManager {
+            verifyDownloadedFileName(firstDownloadedFile)
+            verifyDownloadedFileName(secondDownloadedFile)
+            longClickDownloadedItem(firstDownloadedFile)
+            selectDownloadedItem(secondDownloadedFile)
+            openMultiSelectMoreOptionsMenu()
+            clickMultiSelectRemoveButton()
+            clickSnackbarButton("UNDO")
+            verifyDownloadedFileName(firstDownloadedFile)
+            verifyDownloadedFileName(secondDownloadedFile)
+        }
+    }
+
+    @Test
+    fun systemNotificationCantBeDismissedWhileDownloadingTest() {
+        // Clear the "Firefox Fenix default browser notification"
+        notificationShade {
+            cancelAllShownNotifications()
+        }
+
+        downloadFile = "1GB.zip"
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(downloadTestPage.toUri()) {
+            waitForPageToLoad()
+        }.clickDownloadLink(downloadFile) {
+            verifyDownloadPrompt(downloadFile)
+        }.clickDownload {
+        }
+        browserScreen {
+        }.openNotificationShade {
+            verifySystemNotificationExists("Firefox Fenix")
+            expandNotificationMessage()
+            swipeDownloadNotification("Left", false)
+            verifySystemNotificationExists("Firefox Fenix")
+        }.closeNotificationTray {
+        }.openNotificationShade {
+            verifySystemNotificationExists("Firefox Fenix")
+            expandNotificationMessage()
+            swipeDownloadNotification("Right", false)
+            verifySystemNotificationExists("Firefox Fenix")
+        }
```

Comment:
Does this abort the download afterwards?  I'm cautious about the 1GB.zip downloads.

Reply:
The downloads are aborted once the test finishes, added `deleteDownloadedFileOnStorage` where applicable to also delete the file from the devices storage at the end of the tests.